# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/2d/HelpersD2D.h
# Commit: 5587ed6a3e35
# Full Hash: 5587ed6a3e35161635cecd4c6daf3a3ff8a279cb
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2025-03-01 09:17:11
# Regressor Bug: 1950085
# File Overlap Count: 1
# Description:
#   Bug 1950085 - Ensure DrawTargetWebgl::DrawSurface clips drawing to surface bounds. r=aosmond
#   
#   For offset surfaces, the source rect to DrawSurface may not actually be entirely filled by
#   the surface's contents. In this case, we need to clip the source and dest rects to reflect
#   the actual surface content bounds.
# ==============================================================================

diff -r 9e52dcfa8931 -r 5587ed6a3e35 gfx/2d/HelpersD2D.h
--- a/gfx/2d/HelpersD2D.h	Fri Feb 28 16:06:03 2025 +0000
+++ b/gfx/2d/HelpersD2D.h	Fri Feb 28 18:20:19 2025 +0000
@@ -27,7 +27,8 @@
 
 RefPtr<ID2D1Factory1> D2DFactory();
 
-static inline D2D1_POINT_2F D2DPoint(const Point& aPoint) {
+template <typename T>
+static inline D2D1_POINT_2F D2DPoint(const T& aPoint) {
   return D2D1::Point2F(aPoint.x, aPoint.y);
 }
 
